### PR TITLE
Add "--enable-profiler" flag to all the StackStorm services + CLI

### PR DESCRIPTION
**Pull request by Kami**
This pull request adds ``--enable-profiler`` flag to all the StackStorm services + CLI which can be used to profiler the code using cProf based profiler during development and troubleshooting.

Keep in mind that cProf offers only basic information and usually when drilling down (into a function) you need to use line profiler, but this may not working as expected our of the box due to our usage of eventlet (need to add some docs on that).

### CODE REVIEW

File: st2client/st2client/shell.py [hunk 1]
```diff
@@ -65,7 +65,7 @@
 from st2client.utils.logging import LogLevelFilter, set_log_level_for_all_loggers
 from st2client.commands.auth import TokenCreateCommand
 from st2client.commands.auth import LoginCommand
-
+from st2client.utils.profiler import setup_regular_profiler
```
Comment (Kami) [inline st2client/st2client/shell.py:68]: NOTE: st2client can't depend on st2common that's why we intentionally need to duplicate the code in this package.

**Comment by blag** [comment]
Can we use a different profiler? Something like [Scalene](https://github.com/plasma-umass/scalene)?

**Comment by Kami** [comment]
@blag We probably could for code which doesn't rely on eventlet.

For most rest of the code which relies on eventlet we probably can't use it (unless that profiler knows how to handle eventlet code and context switching).

**Review by cognifloyd — APPROVED** [review]

**Comment by cognifloyd** [comment]
@Kami looks like all you have to do is re-sign the CLA (which we changed recently to say Linux Foundation instead of one of the previous StackStorm stewards).

**Pull request merged**